controllers/fenceagentsremediation_controller.go

```diff
@@ -222,9 +234,22 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 	if meta.IsStatusConditionTrue(far.Status.Conditions, v1alpha1.FenceAgentActionSucceededType) &&
 		!meta.IsStatusConditionTrue(far.Status.Conditions, commonConditions.SucceededType) {
 		// Fence agent action succeeded, and now we try to remove workloads (pods and their volume attachments)
-		r.Log.Info("Manual workload deletion", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
-		if err := utils.DeleteResources(ctx, r.Client, req.Name); err != nil {
-			r.Log.Error(err, "Manual workload deletion has failed", "CR's Name", req.Name)
+		switch far.Spec.RemediationStrategy {
+		case v1alpha1.ResourceDeletionRemediationStrategy:
+			r.Log.Info("Resource deletion - manually deleting workload", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
+			if err := utils.DeleteResources(ctx, r.Client, req.Name); err != nil {
+				r.Log.Error(err, "Resource deletion has failed", "CR's Name", req.Name)
+				return emptyResult, err
+			}
+		case v1alpha1.OutOfServiceTaintRemediationStrategy:
+			r.Log.Info("Resource deletion - adding OutOfService taint", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
```

Comment:
I think it is a bit confusing to state resource deletion for OutOfService taint since the other remediation strategy is called `Resource deletion` (even though the taint causes a resource deletion).

Therefore I suggest `Resource deletion - adding OutOfService taint` -> `Remediation strategy is OutOfService taint which implicitly deletes resources - adding OutOfService taint`.